diff -ur orig.binutils/ld/scripttempl/elf.sc binutils/ld/scripttempl/elf.sc > binutils.patch
--- orig.binutils/ld/scripttempl/elf.sc	2016-09-30 21:06:08.000000000 +0200
+++ binutils/ld/scripttempl/elf.sc	2016-09-30 21:21:06.000000000 +0200
@@ -553,7 +553,8 @@
   ${CREATE_SHLIB-${CREATE_PIE-${RELOCATING+. = ${DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}}
   ${CREATE_SHLIB+${RELOCATING+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}
   ${CREATE_PIE+${RELOCATING+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}
-
+  /* Hacky hackity hack. Need this because otherwise data segment is not aligned enough. */
+  . = ALIGN(0x10000);
   /* Exception handling  */
   .eh_frame     ${RELOCATING-0} : ONLY_IF_RW { KEEP (*(.eh_frame)) ${RELOCATING+*(.eh_frame.*)} }
   .gnu_extab    ${RELOCATING-0} : ONLY_IF_RW { *(.gnu_extab) }
